packages/next/next-server/server/config.ts

```diff
@@ -1,104 +1,18 @@
 import chalk from 'chalk'
 import findUp from 'next/dist/compiled/find-up'
-import os from 'os'
 import { basename, extname } from 'path'
-import { execOnce } from '../lib/utils'
 import * as Log from '../../build/output/log'
 import { CONFIG_FILE } from '../lib/constants'
-import { Header, Rewrite, Redirect } from '../../lib/load-custom-routes'
+import { execOnce } from '../lib/utils'
+import { defaultConfig, normalizeConfig } from './config-shared'
+import { loadWebpackHook } from './config-utils'
 import { ImageConfig, imageConfigDefault, VALID_LOADERS } from './image-config'
 
+export { DomainLocales, NextConfig, normalizeConfig } from './config-shared'
+
 const targets = ['server', 'serverless', 'experimental-serverless-trace']
 const reactModes = ['legacy', 'blocking', 'concurrent']
 
-export type DomainLocales = Array<{
-  http?: true
-  domain: string
-  locales?: string[]
-  defaultLocale: string
-}>
-
-export type NextConfig = { [key: string]: any } & {
-  i18n?: {
-    locales: string[]
-    defaultLocale: string
-    domains?: DomainLocales
-    localeDetection?: false
-  } | null
-
-  headers?: () => Promise<Header[]>
-  rewrites?: () => Promise<Rewrite[]>
-  redirects?: () => Promise<Redirect[]>
-
-  trailingSlash?: boolean
-
-  future: {
-    strictPostcssConfiguration: boolean
-    excludeDefaultMomentLocales: boolean
-    webpack5: boolean
-  }
-}
-
-const defaultConfig: NextConfig = {
```

Comment:
This removal just invalidated the docs, which shouldn't've been pointed to `canary` anyway: https://nextjs.org/docs/api-reference/next.config.js/introduction

Reply:
Could you elaborate more? Those docs are still accurate!